File: CS/WindowsApplication1/Form1.cs
using System;
using System.Data;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        Random random = new Random();

        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            gridControl1.ShowPrintPreview();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            productsBindingSource.DataSource = GetProductsDataTable();
        }

        DataTable GetProductsDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Products";
            table.Columns.Add(new DataColumn("ProductName", typeof(string)));
            table.Columns.Add(new DataColumn("UnitPrice", typeof(double)));
            table.Columns.Add(new DataColumn("UnitsOnOrder", typeof(int)));
            for(int i = 0; i < 20; i++)
            {
                int index = i + 1;
                table.Rows.Add("Product " + index, Math.Round(random.NextDouble() * 1000, 2), random.Next(0, 9) * 10);
            }
            return table;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form

		Private random As New Random()

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			gridControl1.ShowPrintPreview()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			productsBindingSource.DataSource = GetProductsDataTable()
		End Sub

		Private Function GetProductsDataTable() As DataTable
			Dim table As New DataTable()
			table.TableName = "Products"
			table.Columns.Add(New DataColumn("ProductName", GetType(String)))
			table.Columns.Add(New DataColumn("UnitPrice", GetType(Double)))
			table.Columns.Add(New DataColumn("UnitsOnOrder", GetType(Integer)))
			For i As Integer = 0 To 19
				Dim index As Integer = i + 1
				table.Rows.Add("Product " & index, Math.Round(random.NextDouble() * 1000, 2), random.Next(0, 9) * 10)
			Next i
			Return table
		End Function
	End Class
End Namespace